app/models/user.rb

```diff
@@ -93,7 +93,7 @@ def is_unclaimed?
   end
 
   def is_email_pending?
-    !email_addresses.any?(&:verified)
+    !email_addresses.any?(&:verified) && !email_addresses.empty?
   end
```

Comment:
obviously my "sounds good to me" was meant to be about your last two comments :-)